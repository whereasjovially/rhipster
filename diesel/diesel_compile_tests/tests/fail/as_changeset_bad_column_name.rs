#[macro_use]
extern crate diesel;

table! {
    users {
        id -> Integer,
    }
}

#[derive(AsChangeset)]
#[table_name = "users"]
struct UserStruct {
    name: String,
    #[column_name = "hair_color"]
    color_de_pelo: String,
}

#[derive(AsChangeset)]
#[table_name = "users"]
struct UserTuple(#[column_name = "name"] String);

fn main() {}
