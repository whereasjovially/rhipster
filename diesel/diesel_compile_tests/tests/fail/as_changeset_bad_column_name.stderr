error[E0412]: cannot find type `name` in module `users`
  --> $DIR/as_changeset_bad_column_name.rs:13:5
   |
13 |     name: String,
   |     ^^^^ not found in `users`

error[E0412]: cannot find type `hair_color` in module `users`
  --> $DIR/as_changeset_bad_column_name.rs:14:21
   |
14 |     #[column_name = "hair_color"]
   |                     ^^^^^^^^^^^^ not found in `users`
   |
   = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0425]: cannot find value `name` in module `users`
  --> $DIR/as_changeset_bad_column_name.rs:13:5
   |
13 |     name: String,
   |     ^^^^ not found in `users`

error[E0425]: cannot find value `hair_color` in module `users`
  --> $DIR/as_changeset_bad_column_name.rs:14:21
   |
14 |     #[column_name = "hair_color"]
   |                     ^^^^^^^^^^^^ not found in `users`
   |
   = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0412]: cannot find type `name` in module `users`
  --> $DIR/as_changeset_bad_column_name.rs:20:34
   |
20 | struct UserTuple(#[column_name = "name"] String);
   |                                  ^^^^^^ not found in `users`
   |
   = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0425]: cannot find value `name` in module `users`
  --> $DIR/as_changeset_bad_column_name.rs:20:34
   |
20 | struct UserTuple(#[column_name = "name"] String);
   |                                  ^^^^^^ not found in `users`
   |
   = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)
